packages/server/test/e2e/7_issue_15901_spec.js

```diff
@@ -0,0 +1,14 @@
+const e2e = require('../support/helpers/e2e').default
+const Fixtures = require('../support/helpers/fixtures')
+
+// https://github.com/cypress-io/cypress/issues/15901
+describe('issue 15901', () => {
+  e2e.setup()
+
+  it('does not crash', function () {
+    return e2e.exec(this, {
+      project: Fixtures.projectPath('issue-15901'),
+      expectedExitCode: 0,
+    })
+  })
+})
```

Comment:
This test looks good in general, but can you move it to `net_stubbing_spec` in `packages/driver`? It doesn't need to be an e2e test.

Reply:
I first wrote this test as an integration test, see the commits BUT the request buffer did not arrive in the same binary buffer as a "normal" request. Probably because the middleware during our integration tests processes the requests differently. Thus to recreate the crash before the fix, had to use e2e test

Reply:
Hm, then it sounds like it isn't the right reproduction, because there should not be any difference between cy.intercept inside our integration tests vs. outside. If you need to setup server-side code in driver to test it properly you can modify `packages/driver/cypress/plugins/server.js`.

Reply:
I did add it into the server, but could not get the crash, so in order to make sure we get the same behavior as the user, e2e seems like a good test

Reply:
Why was it not crashing in that context? It seems worth knowing that at least. Can you investigate this and fix the test to be a driver test? We are trying not to add new unnecessary/issue-specific e2e tests.

Reply:
:(

that feels like extra work that does not benefit the users but rather puts more complexity into our integration setup

Reply:
One of our goals for the future is to consolidate our existing e2e tests and remove unneeded tests and convert them to more component-specific tests. To that end, we should avoid unnecessary e2e tests.

Also, if you don't know why it didn't crash inside of the integration test, doesn't that indicate a potential issue with the fix...?

Reply:
Moved the test into driver integration and it seems to work.